internal/cli/runner_install.go

```diff
@@ -0,0 +1,269 @@
+			Target:  &c.runnerProfileOdrImage,
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "server-tls",
+			Target:  &c.serverTls,
+			Usage:   "If true, the Waypoint runner will connect to the server over TLS.",
+			Default: true,
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:   "server-tls-skip-verify",
+			Target: &c.serverTlsSkipVerify,
+			Usage:  "If true, will not validate TLS cert presented by the server.",
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:   "server-require-auth",
+			Target: &c.serverRequireAuth,
+			Usage:  "If true, will send authentication details.",
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "adopt",
+			Usage:   "Adopt the runner after it is installed.",
+			Default: true,
+			Target:  &c.adopt,
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "id",
+			Usage:  "If this is set, the runner will use the specified id.",
+			Target: &c.id,
+		})
+
+		sort.Strings(runnerPlatforms)
+
+		for _, name := range runnerPlatforms {
+			platform := runnerinstall.Platforms[name]
+			platformSet := set.NewSet(name + " Options")
+			platform.InstallFlags(platformSet)
+		}
+	})
+}
+
+func (c *RunnerInstallCommand) Help() string {
+	return formatHelp(`
+Usage: waypoint runner install [options]
+
+  Install a Waypoint runner to an existing platform. The platform should be
+  specified as kubernetes, nomad, ecs, or docker.
+
+  With the '-adopt' flag set to true (the default), this command will adopt the
+  runner after it is installed. The install will attempt to install a runner for
+  the server configured in the current Waypoint context.
+
+` + c.Flags().Help())
+}
+
+func (c *RunnerInstallCommand) Synopsis() string {
+	return "Install a Waypoint runner to Kubernetes, Nomad, ECS, or Docker"
+}
+
+func (c *RunnerInstallCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithNoLocalServer(), // no auth in local mode
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+
+	ctx := c.Ctx
+	log := c.Log.Named("install")
+	defer c.Close()
+
+	if len(c.platform) == 0 {
+		c.ui.Output(
+			"The -platform flag is required.",
+			terminal.WithErrorStyle(),
+		)
+		c.ui.Output(c.Flags().Help(), terminal.WithInfoStyle())
+		return 1
+	}
+	p, ok := runnerinstall.Platforms[strings.ToLower(c.platform[0])]
+	if !ok {
+		c.ui.Output(
+			"Error installing server into %q: unsupported platform",
+			c.platform,
+			terminal.WithErrorStyle(),
+		)
+
+		return 1
+	}
+
+	sg := c.ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Connecting to: %s", c.serverUrl)
+	defer func() { s.Abort() }()
+
+	var cookie string
+	if c.adopt {
+		serverConfig, err := c.project.Client().GetServerConfig(ctx, &empty.Empty{})
+		if err != nil {
+			c.ui.Output(
+				"Error getting server config.",
+				clierrors.Humanize(err),
+				terminal.WithErrorStyle(),
+			)
+			return 1
+		}
+		cookie = serverConfig.Config.Cookie
+	}
+
+	if c.serverUrl == "" {
```

Comment:
I think like the platform, we can get the server address from context. The current `runner agent` command does this.